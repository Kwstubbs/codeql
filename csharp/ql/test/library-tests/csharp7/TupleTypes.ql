import csharp

from TupleType tt, int i
where tt.fromSource()
select tt.getName(), tt.toStringWithTypes(), tt.getUnderlyingType().toStringWithTypes(),
  tt.getArity(), i, tt.getElement(i).getName()
