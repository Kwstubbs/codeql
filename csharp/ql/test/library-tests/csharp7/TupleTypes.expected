| (Int32,(String,Int32)) | (int, (string, int)) | ValueTuple<int, (string, int)> | 2 | 0 | Item1 |
| (Int32,(String,Int32)) | (int, (string, int)) | ValueTuple<int, (string, int)> | 2 | 1 | Item2 |
| (Int32,Double) | (int, double) | ValueTuple<int, double> | 2 | 0 | Item1 |
| (Int32,Double) | (int, double) | ValueTuple<int, double> | 2 | 1 | Item2 |
| (Int32,String) | (int, string) | ValueTuple<int, string> | 2 | 0 | Item1 |
| (Int32,String) | (int, string) | ValueTuple<int, string> | 2 | 1 | Item2 |
| (String,String) | (string, string) | ValueTuple<string, string> | 2 | 0 | Item1 |
| (String,String) | (string, string) | ValueTuple<string, string> | 2 | 1 | Item2 |
